### ISSUE TITLE
llama rope sin cos的形状

### ISSUE THREAD

**Issue by feifeibear** [ModelTC/lightllm#319]
我发现你们框架中，cos和sin的最后一个维度是dim//2，这个和其他的不太一样，一般是dim。我想请教一下这样设计的原因？
比如huggingface的就是(batch, seq_len, 1, dim)，计算q*cos + q * sin
https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L211

你们的triton kernel应该对应如下实现，和hf的是不是有差别？
```python
def rotary_emb_fwd(x, cos, sin):
    """
    inline update x
    """
    seq_len, h, dim = x.shape
    x0 = x[:, :, 0: dim // 2]
    x1 = x[:, :, dim // 2: dim]
    cos = cos.view((seq_len, 1, dim // 2))
    sin = sin.view((seq_len, 1, dim // 2))
    o0 = x0 * cos - x1 * sin
    o1 = x0 * sin + x1 * cos
    res = torch.cat((o0, o1), dim=-1)
    x.copy_(res)
    return
```

**Comment by shihaobai** [comment]
https://github.com/huggingface/transformers/blob/963db81a5afc5c31edafa0c9fb0390956a78cf2a/src/transformers/models/llama/modeling_llama.py#L204 可以参考这个函数，我们的实现跟他是等价的
